File: Lab2.2cs/Figure.cs
using System;

namespace Lab2_2
{
    public abstract class Figure
    {
        private string _name;

        public string Name
        {
            get { return _name; }
        }

        public Figure()
        {
            _name = "Unknown Figure";
        }

        public Figure(string name)
        {
            _name = name;
        }

        public virtual double CalculateArea()
        {
            return 0;
        }

        public virtual double CalculatePerimeter()
        {
            return 0;
        }

        public override string ToString()
        {
            return $"Figure: {_name}";
        }
    }
}

File: Lab2.2/Figure.cpp
#include "Figure.h"

Figure::Figure() : name("Unknown Figure") {
}

Figure::Figure(const std::string& name) : name(name) {
}

Figure::Figure(const Figure& other) : name(other.name) {
}

Figure::~Figure() {
}

std::string Figure::getName() const {
    return name;
}

double Figure::calculateArea() const {
    return 0.0;
}

double Figure::calculatePerimeter() const {
    return 0.0;
}

std::string Figure::toString() const {
    return "Figure: " + name;
}
